### PR TITLE
Adding tests and fixing `matrix_sparseSandwitch` function

### PR DESCRIPTION
**Pull request by PNieck**
## Description

This PR fixes `matrix_sparseSandwitch` function and adds tests for this function.

## Motivation and Context

Better interface in `matrix` library and more precise tests.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [x] New test added: (add PR link here). In this PR
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [x] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: algebra/tests/matrix/sandwitch.c [hunk 1]
```diff
@@ -431,3 +431,412 @@ TEST_GROUP_RUNNER(group_matrix_sandwitch)
...
+TEST_TEAR_DOWN(group_matrix_sparseSandwitch_stdMat)
+{
+	matrix_bufFree(&M1);
+	matrix_bufFree(&M2);
+	matrix_bufFree(&M3);
+	matrix_bufFree(&Exp);
+	matrix_bufFree(&tmp);
+}
+
+#include <stdio.h>
```
Comment (niewim19) [inline algebra/tests/matrix/sandwitch.c:475]: Debugging reminiscence

File: algebra/tests/matrix/sandwitch.c [hunk 2]
```diff
@@ -431,3 +431,412 @@ TEST_GROUP_RUNNER(group_matrix_sandwitch)
...
+TEST(group_matrix_sparseSandwitch_stdMat, matrix_sparseSandwitch_std)
+{
+	TEST_ASSERT_EQUAL_INT(SANDWITCH_OK, matrix_sparseSandwitch(&M1, &M2, &M3, &tmp));
+
+	matrix_print(&M3);
+	printf("------\n");
+	matrix_print(&Exp);
```
Comment (niewim19) [inline algebra/tests/matrix/sandwitch.c:482]: Debugging reminiscence

File: algebra/tests/matrix/sandwitch.c [hunk 3]
```diff
@@ -431,3 +431,412 @@ TEST_GROUP_RUNNER(group_matrix_sandwitch)
...
+	TEST_ASSERT_EQUAL_INT(SANDWITCH_FAIL, matrix_sparseSandwitch(&M1, &M2, &M3, &tmp));
+
+	matrix_trp(&M3);
+
+	TEST_ASSERT_EQUAL_INT(SANDWITCH_FAIL, matrix_sparseSandwitch(&M1, &M2, &M3, &tmp));
+}
+
+
+/* This test checks if result matrix and temporary are changing when function fails */
+TEST(group_matrix_sparseSandwitch_badMats, matrix_SANDWITCH_FAILureRetain)
```
Comment (niewim19) [inline algebra/tests/matrix/sandwitch.c:801]: Looks like copy-paste error of upper case

**Review by niewim19 — CHANGES_REQUESTED** [review]

File: algebra/tests/matrix/sandwitch.c (hunk 3 above)
Comment (PNieck) [reply algebra/tests/matrix/sandwitch.c:801]: No, it is a mistake made by *find and replace* action ;)

**Review by niewim19 — APPROVED** [review]
LGTM

**Pull request merged**